clients/slo/slo.go

```diff
@@ -0,0 +1,212 @@
+	if id == "" {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, errors.New("argument \"id\" is empty"))
+	}
+
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	version, err := getOptimisticLockingVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+
+	resp, err := c.restClient.PUT(ctx, path, bytes.NewReader(body), rest.RequestOptions{
+		CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
+		QueryParams:           url.Values{"optimistic-locking-version": []string{version}},
+	})
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "update", id, err)
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+func (c *Client) Delete(ctx context.Context, id string) (api.Response, error) {
+	if id == "" {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, errors.New("argument \"id\" is empty"))
+	}
+
+	getResp, err := c.Get(ctx, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	version, err := getOptimisticLockingVersion(getResp)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	path, err := url.JoinPath(endpointPath, id)
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+
+	resp, err := c.restClient.DELETE(ctx, path, rest.RequestOptions{
+		CustomShouldRetryFunc: rest.RetryIfTooManyRequests,
+		QueryParams:           url.Values{"optimistic-locking-version": []string{version}},
+	})
+	if err != nil {
+		return api.Response{}, fmt.Errorf(errMsgWithId, "delete", id, err)
+	}
+	defer resp.Body.Close()
+
+	return api.ProcessResponse(resp)
+}
+
+const errMsg = "failed to %s slo resource: %w"
+const errMsgWithId = "failed to %s slo resource with id %s: %w"
+
+func getOptimisticLockingVersion(resp api.Response) (string, error) {
+	var body struct {
+		Version string `json:"version"`
+	}
+
+	err := json.Unmarshal(resp.Data, &body)
+	if err != nil {
+		return "", err
+	}
+
+	return body.Version, nil
+}
+
+func processListResponse(httpResponse *http.Response) (string, api.ListResponse, error) {
+	var err error
+
+	var body json.RawMessage
+	if body, err = io.ReadAll(httpResponse.Body); err != nil {
+		return "", api.ListResponse{}, api.NewAPIErrorFromResponse(httpResponse)
+	}
+
+	if !rest.IsSuccess(httpResponse) {
+		return "", api.ListResponse{}, api.NewAPIErrorFromResponseAndBody(httpResponse, body)
+	}
+
+	var s struct {
+		NextPage string            `json:"nextPageKey"`
+		Data     []json.RawMessage `json:"slos"`
+	}
+
+	if err := json.Unmarshal(body, &s); err != nil {
+		return "", api.ListResponse{}, api.NewAPIErrorFromResponseAndBody(httpResponse, body)
```

Comment:
Error is completely lost why the unmarshalling failed

Reply:
This is C/P. The same situation is on ln 189-190. I would leave it like this: Let's discuss what the desired behavior is and change it for all clients.